lib/wallet/wallet.js

```diff
@@ -1561,7 +1558,17 @@ class Wallet extends EventEmitter {
     if (start !== 0 && start !== height)
       throw new Error('Name is already opening.');
 
-    const addr = await this.receiveAddress(acct);
+    // Send 0-value output to provably unspendable address:
+    // hs1lqqqqhuxwgy (mainnet) is the smallest valid nulldata address
+    // and represents witness version 31 folowed by data of 0x0000.
+    // Unspendable outputs do not bloat the UTXO set or the wallet's coin set.
+    const addr = Address.fromNulldata(Buffer.alloc(2));
+
+    // Since this address is not in our wallet, we must "manually" add the name
+    // to the wallet database's name map. Normally the wallet would do this in
+    // txdb.js connectNames() when a TX is inserted into the DB, and the wallet
+    // recognizes the output address.
+    await this.importName(name);
```

Comment:
Maybe I missed it,

but does not this need to change rescan logic as well, so wallets can recover the OPENs they have sent?

Reply:
`importName()` will add the name to the wallet's name map, which means the wallet will "track" all auction activity like bids etc even if the user did not bid themselves (this is weird, but its how the wallet works already). This will persist for rescans as well, especially because of this change: https://github.com/handshake-org/hsd/pull/499/commits/0301f47e6163b55712c2bd97af4adb54790bfed7

Since the wallet _sent_ the open, it should recover the TX normally because its own keys were used for the inputs and change.

OPENs are weird things - they are just a marker for the chain and wallet (this is why they have 0 value), announcing to the network a namehash and its preimage. JJ told me OPENs weren't even added to the system until late in development because they didn't seem necessary, but definitely made wallet logic a lot easier.

We have a few rescan tests now in hsd, I'll rebase this PR and we can double check

Reply:
sure,

I was wondering about importing wallet into fresh wallet instance.
I can test after rebase.

Reply:
> I was wondering about importing wallet into fresh wallet instance.

Yes! this needs to be patched. Its' going to be in `txdb.insert()` or `txdb.confirm()` I think